packages/renderer/src/lib/preferences/PreferencesRegistriesEditing.svelte

```diff
@@ -0,0 +1,459 @@
+<style>
+.dropdown:hover .dropdown-menu {
+  display: block;
+}
+</style>
+
+<script lang="ts">
+import type { Registry } from '@tmpwip/extension-api';
+import { onMount } from 'svelte';
+import { registriesInfos } from '../../stores/registries';
+
+let registryStore: Registry[] = [];
+onMount(() => {
+  registriesInfos.subscribe(value => {
+    registryStore = value;
+  });
+});
+
+let dirtyRegistryStore: Registry[] = [];
+let loginErrorMessages: { registry: Registry; error: string }[] = [];
+
+$: isRegistryDirty = function isRegistryDirty(registry: Registry): boolean {
+  return dirtyRegistryStore.find(r => r.serverUrl === registry.serverUrl) !== undefined;
+};
+
+function markRegistryAsDirty(registry: Registry) {
+  hidePasswordForRegistry(registry);
+
+  const dirtyRegistry = {
+    source: registry.source,
+    serverUrl: registry.serverUrl,
+    username: registry.username,
+    secret: registry.secret,
+  } as Registry;
+
+  dirtyRegistryStore.push(dirtyRegistry);
+
+  updateUI();
+}
+
+function markRegistryAsClean(registry: Registry) {
+  const index = dirtyRegistryStore.findIndex(r => r.serverUrl === registry.serverUrl, 0);
+  if (index > -1) {
+    dirtyRegistryStore.splice(index, 1);
+  }
+
+  updateUI();
+}
+
+$: getDirtyRegistry = function getDirtyRegistry(registry: Registry): Registry | undefined {
+  return dirtyRegistryStore.find(r => r.serverUrl === registry.serverUrl);
+};
```

Comment:
Hi, I'm not sure to follow what is a dirty registry and why this function is mapped to a reactive call ?

Reply:
removed reactive declaration